llvm/lib/CodeGen/MachineScheduler.cpp

```diff
@@ -3897,6 +3897,12 @@ void PostGenericScheduler::initialize(ScheduleDAGMI *Dag) {
 void PostGenericScheduler::initPolicy(MachineBasicBlock::iterator Begin,
                                       MachineBasicBlock::iterator End,
                                       unsigned NumRegionInstrs) {
+  const MachineFunction &MF = *Begin->getMF();
+
+  // Allow the subtarget to override default policy.
+  MF.getSubtarget().overridePostRASchedPolicy(RegionPolicy, NumRegionInstrs);
+
+  // After subtarget overrides, apply command line options.
   if (PostRADirection == MISchedPostRASched::TopDown) {
     RegionPolicy.OnlyTopDown = true;
     RegionPolicy.OnlyBottomUp = false;
```

Comment:
This always overrides what the subtarget wanted for these two members because `PostRADirection` defaults to `TopDown`. Maybe add a `getNumOccurrences()` check, or make the direction enum have an explicit default entry?

Reply:
Thanks! I didn't notice this mistake.
Done. I added a `getNumOccurrences()` check now, what do you think about current code?